pkg/controller/license/license_controller.go

```diff
@@ -198,19 +198,12 @@ func (r *ReconcileLicenses) reconcileClusterLicense(cluster esv1.Elasticsearch)
 	}
 	matchingSpec, parent, found := findLicense(r, r.checker, minVersion)
 	if !found {
-		// no license, delete cluster level licenses to revert to basic
-		log.V(1).Info("No enterprise license found. Attempting to remove cluster license secret", "namespace", cluster.Namespace, "es_name", cluster.Name)
-		secretName := esv1.LicenseSecretName(cluster.Name)
-		err := r.Client.Delete(context.Background(), &corev1.Secret{
-			ObjectMeta: k8s.ToObjectMeta(types.NamespacedName{
-				Namespace: cluster.Namespace,
-				Name:      secretName,
-			}),
+		// no matching license found, delete cluster level license if it exists to revert to basic
+		clusterLicenseNSN := types.NamespacedName{Namespace: cluster.Namespace, Name: esv1.LicenseSecretName(cluster.Name)}
+		err := k8s.DeleteSecretIfExists(r.Client, clusterLicenseNSN, func() {
+			log.V(1).Info("No enterprise license found. Attempting to remove cluster license secret", "namespace", cluster.Namespace, "es_name", cluster.Name)
```

Comment:
Is it really important to log this between the `get` and the `delete`. Otherwise, we keep the log before `DeleteSecretIfExists` and we can remove the `onDelete` callback.

Reply:
Sure I am open to that.